dockerfiles/feature-support/hsm-patches/0001-Introduce-generic-keypair-generation-interface-and-e.patch

```diff
@@ -343,16 +344,31 @@ index ec7f279..8466700 100644
  
  	PKCS11_CTX_private *ctx = slot->ctx;
  	CK_SESSION_HANDLE session;
-@@ -266,8 +266,6 @@ int pkcs11_generate_key(PKCS11_SLOT_private *slot, int algorithm, unsigned int b
+@@ -266,10 +266,20 @@ int pkcs11_generate_key(PKCS11_SLOT_private *slot, int algorithm, unsigned int b
  	CK_OBJECT_HANDLE pub_key_obj, priv_key_obj;
  	int rv;
  
 -	(void)algorithm; /* squash the unused parameter warning */
 -
- 	if (pkcs11_get_session(slot, 1, &session))
+-	if (pkcs11_get_session(slot, 1, &session))
++	// R/W session is mandatory for key generation.
++	if (pkcs11_open_session(slot, 1)) {
```

Comment:
1 is a magic number. Do we want to do "#define RW 1" somewhere?
```suggestion
+	if (pkcs11_open_session(slot, RW)) {
```
?

Reply:
This is how it's done across all of libp11's codebase :(